src/Nox.Lib/Middlewares/NoxApiMiddleware.cs

```diff
@@ -0,0 +1,167 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Template;
+using Nox.Solution;
+using System.Diagnostics;
+
+namespace Nox.Lib;
+
+public class NoxApiMiddleware
+{
+    private readonly PathString _apiPrefix;
+
+    private readonly RequestDelegate _next;
+
+    private readonly List<RouteMatcher> _matchers = new();
+
+    public NoxApiMiddleware(RequestDelegate next, NoxSolution solution)
+    {
+        _next = next;
+
+        _apiPrefix = solution.Presentation.ApiConfiguration.ApiRoutePrefix;
+
+        foreach (var route in solution.Presentation.ApiConfiguration.ApiRouteMappings)
+        {
+            _matchers.Add(new RouteMatcher(route, _apiPrefix));
+        }
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        var path = context.Request.Path;
+        var verb = context.Request.Method;
+
+        if (!path.HasValue)
+        {
+            await _next(context);
+            return;
+        }
+
+        if (!path.StartsWithSegments(_apiPrefix))
+        {
+            await _next(context);
+            return;
+        }
+
+        RouteValueDictionary? urlParameters = null;
+
+        ApiRouteMapping? apiRoute = null;
+
+        foreach (var matcher in _matchers)
+        {
+            urlParameters = matcher.Match(path);
+
+            if (urlParameters != null)
+            {
+                apiRoute = matcher.ApiRoute;
+                break;
+            }
+        }
+
+        if (urlParameters == null || apiRoute == null)
+        {
+            await _next(context);
+            return;
+        }
+
+        Debug.WriteLine(context.Request.Path.Value);
+
+        if (!apiRoute.HttpVerb.ToString().ToUpper().Equals(verb))
```

Comment:
for performance improvements?
```suggestion
        if(!string.Equals(apiRoute.HttpVerb.ToString(), verb, StringComparison.OrdinalIgnoreCase)
```